### PR TITLE
Vtk mapped properties II

### PR DESCRIPTION
**Pull request by TomFischer**
This PR contains missing commits from the **very old** PR https://github.com/ufz/ogs/pull/607 which can't easily be rebased. For this reason @bilke and I compared most of the commits with the current state of the master and applied commits manually. I hope we could recover all the things introduced into the previous PR. Unfortunately, there are not any tests. For this reason, @rinkk please have a closer look to the DataExplorer parts!

### CODE REVIEW

File: Applications/DataExplorer/DataView/MshModel.cpp [hunk 1]
```diff
@@ -63,8 +62,8 @@ void MshModel::addMeshObject(const MeshLib::Mesh* mesh)
 	QList<QVariant> meshData;
 	meshData << display_name << "";
 	MshItem* newMesh = new MshItem(meshData, _rootItem, mesh);
-	if (newMesh->vtkSource())
-		newMesh->vtkSource()->SetName(display_name);
+	//if (newMesh->vtkSource())
+	//	newMesh->vtkSource()->SetName(display_name);
```
Comment (endJunction) [inline Applications/DataExplorer/DataView/MshModel.cpp]: Please add a comment why this is not working or not executed. (or remove entirely).

**Comment by TomFischer** [comment]
The issues mentioned in the comments
> could be a unique_ptr for enhanced memory leak protection.
> std::
> 2012-2015

are solved with later commits already.

File: Applications/DataExplorer/DataView/MshModel.cpp (hunk 1 above)
Comment (TomFischer) [inline Applications/DataExplorer/DataView/MshModel.cpp]: Added explanation above of the commented lines.

**Comment by endJunction** [comment]
:fast_forward:

**Pull request merged**